File: Node.py
from datetime import datetime
class Node():
    def __init__(self,name,path,children):
        def validateFileName(fileName):
            PROHIBITED = ['<','>',':','"','/','|','\\','?','*']
            for x in PROHIBITED:
                if x in fileName:
                    return False
            return True
        if validateFileName:
            self.name = name
        else:
            raise ValueError('File names must not include any of these symbols <>:"/\|?*')
        self.path = path
        self.data = ''
        self.type = 'FILE' if '.' in name else 'FOLDER'
        self.createdDate = datetime.now()
        self.editedDate = datetime.now()
        self.children = children

File: filesystem/src/fileClasses/Node.ts
export default class Node {
    name: string;
    path: string;
    data: string;
    type: string;
    createdDate: Date;
    editedDate: Date;
    children: Node[];

    constructor(name: string, path: string, children: Node[]) {
        this.name = '';
        if (this.validateFileName(name)) {
            this.name = name;
        } else {
            throw new Error('File names must not include any of these symbols < > : " / \\ | ? *');
        }
        this.path = path;
        this.data = '';
        this.type = name.includes('.') ? 'FILE' : 'FOLDER';
        this.createdDate = new Date();
        this.editedDate = new Date();
        this.children = children;
    }

    validateFileName(fileName: string): boolean {
        const PROHIBITED = ['<', '>', ':', '"', '/', '\\', '|', '?', '*'];
        return !PROHIBITED.some((symbol) => fileName.includes(symbol));
    }
}